aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunClientFactories.java

```diff
@@ -36,12 +37,10 @@ public static AliyunClientFactory defaultFactory() {
     return ALIYUN_CLIENT_FACTORY_DEFAULT;
   }
 
-  public static AliyunClientFactory load(Map<String, String> properties) {
-    if (properties.containsKey(AliyunProperties.CLIENT_FACTORY)) {
-      return load(properties.get(AliyunProperties.CLIENT_FACTORY), properties);
-    } else {
-      return defaultFactory();
-    }
+  public static AliyunClientFactory from(Map<String, String> properties) {
+    String factoryImpl = PropertyUtil.propertyAsString(
+        properties, AliyunProperties.CLIENT_FACTORY, DefaultAliyunClientFactory.class.getName());
```

Comment:
I think the previous version has a small bug, because the `DefaultAliyunClientFactory` forget to initialize the aliyun properties in the constructors.  This PR unified the default aliyun client factory and customized client factory,  it's great to avoid the the properties initialization !